src/components/expression/expression-loader.js

```diff
@@ -211,18 +212,26 @@ genotet.ExpressionLoader.prototype.update = function(method, fileName,
  * @param {!Array<string>} originalNames Original names from previous zoom
  *      status.
  * @param {!Array<string>} removeNames Names need to be removed.
+ * @return {!Array<string>} result Revised names.
  * @private
  */
 genotet.ExpressionLoader.prototype.removeNames_ = function(originalNames,
                                                            removeNames) {
   var namesDict = {};
+  var result = [];
   originalNames.forEach(function(name, i) {
     namesDict[name] = i;
   });
   removeNames.forEach(function(name) {
     var geneIndex = namesDict[name];
     if (name in namesDict) {
-      originalNames.splice(geneIndex, 1);
+      delete originalNames[geneIndex];
```

Comment:
You should not delete array element using index. If you want to unset it, use explicit value assignment.